packages/backend-core/src/context/mainContext.ts

```diff
@@ -122,10 +122,10 @@ export async function doInAutomationContext<T>(params: {
   automationId: string
   task: () => T
 }): Promise<T> {
-  const tenantId = getTenantIDFromAppID(params.appId)
+  await ensureSnippetContext()
```

Comment:
Since snippets always need to be part of automation context, it feels correct to me to add the snippets into context here. Since snippet context works by mutating the current context rather than creating a new layer, we aren't creating 2 new contexts or anything by doing this.